### ISSUE TITLE
non-line mod without path attribute false positive

### ISSUE THREAD

**Issue by matklad** [rust-lang/rust#36772]
This code

```Rust
fn main() {
    #[path="foo"]
    mod foo {
        mod bar;
    }
}
```

Fails to compile because there is no path on the `bar`, however I think it should work because there is a path on the `foo`. That is, the following is valid and will look for `foo/bar.rs`:

```Rust
fn main() {}

#[path="foo"]
mod foo {
    mod bar;
}
```

cc @jseyfried https://github.com/rust-lang/rust/commit/d21e908175de9c70fffc9c07d676557392cf2b39
